contrib/drivers/ceph/ceph_metrics.go

```diff
@@ -0,0 +1,192 @@
+}
+
+func metricInMetrics(metric string, metriclist []string) bool {
+	for _, m := range metriclist {
+		if m == metric {
+			return true
+		}
+	}
+	return false
+}
+func getCurrentUnixTimestamp() int64 {
+	now := time.Now()
+	secs := now.Unix()
+	return secs
+}
+func getMetricToUnitMap() map[string]string {
+
+	//construct metrics to value map
+	var configs Configs
+	//Read supported metric list from yaml config
+	//Todo: Move this to read from file
+	source := []byte(data)
+
+	error := yaml.Unmarshal(source, &configs)
+	if error != nil {
+		log.Fatalf("Unmarshal error: %v", error)
+	}
+	metricToUnitMap := make(map[string]string)
+	for _, resources := range configs.Cfgs {
+		switch resources.Resource {
+		//ToDo: Other Cases needs to be added
+		case "pool":
+			for index, metricName := range resources.Metrics {
+
+				metricToUnitMap[metricName] = resources.Units[index]
+
+			}
+		}
+	}
+	return metricToUnitMap
+}
+
+// 	ValidateMetricsSupportList:- is  to check whether the posted metric list is in the uspport list of this driver
+// 	metricList-> Posted metric list
+//	supportedMetrics -> list of supported metrics
+func (d *MetricDriver) ValidateMetricsSupportList(metricList []string, resourceType string) (supportedMetrics []string, err error) {
+	var configs Configs
+
+	//Read supported metric list from yaml config
+	//Todo: Move this to read from file
+	source := []byte(data)
+	error := yaml.Unmarshal(source, &configs)
+	if error != nil {
+		log.Fatalf("Unmarshal error: %v", error)
+	}
+
+	for _, resources := range configs.Cfgs {
+		switch resources.Resource {
+		//ToDo: Other Cases needs to be added
+		case "pool":
+			for _, metricName := range metricList {
+				if metricInMetrics(metricName, resources.Metrics) {
+					supportedMetrics = append(supportedMetrics, metricName)
+
+				} else {
+					log.Infof("metric:%s is not in the supported list", metricName)
+				}
+			}
+		}
+	}
+	return supportedMetrics, nil
+}
+
+//	CollectMetrics: Driver entry point to collect metrics. This will be invoked by the dock
+//	metricsList-> posted metric list
+//	instanceID -> posted instanceID
+//	metricArray	-> the array of metrics to be returned
+func (d *MetricDriver) CollectMetrics(metricsList []string, instanceID string) ([]*model.MetricSpec, error) {
+
+	// get Metrics to unit map
+	metricToUnitMap := getMetricToUnitMap()
+	//validate metric support list
+	supportedMetrics, err := d.ValidateMetricsSupportList(metricsList, "pool")
+	if supportedMetrics == nil {
+		log.Infof("No metrics found in the  supported metric list")
+	}
+	metricMap, err := CollectMetrics(supportedMetrics, instanceID)
+
+	var tempMetricArray []*model.MetricSpec
+	for label_val, _ := range metricMap {
+		for _, element := range supportedMetrics {
+			val, _ := strconv.ParseFloat(metricMap[label_val][element], 64)
+			//Todo: See if association  is required here, resource discovery could fill this information
+			associatorMap := make(map[string]string)
```

Comment:
What if more lables are required for other resource? better to desperate label filling out of metricspec filling logic